### PR TITLE
Validate on the group of post types before bailing

### PR DESCRIPTION
**Pull request by mitogh**
**Ticket** https://central.tri.be/issues/99550

Make sures the event consider all the post types created by the plugin
instead of only the main onoe.

### CODE REVIEW

File: src/Tribe/Query.php [hunk 1]
```diff
@@ -73,8 +73,18 @@ public static function parse_query( $query ) {
 				$query->set( 'paged', $_REQUEST['tribe_paged'] );
 			}
 
-			// Return early as we don't want to change a post that is not an event.
-			if ( $query->is_main_query() && $query->is_single() && $query->get( 'post_type' ) !== Tribe__Events__Main::POSTTYPE )  {
+			// Return early as we don't want to change a post that is not part of the valid group of event post types.
+			$valid_post_types = [
+				Tribe__Events__Main::POSTTYPE,
+				Tribe__Events__Main::ORGANIZER_POST_TYPE,
```
Comment (bordoni) [inline src/Tribe/Query.php]: We want to use: `Tribe__Events__Venue::POSTTYPE` and `Tribe__Events__Organizer::POSTTYPE`

File: src/Tribe/Query.php [hunk 2]
```diff
@@ -73,8 +73,18 @@ public static function parse_query( $query ) {
...
+				Tribe__Events__Main::VENUE_POST_TYPE,
```
Comment (bordoni) [inline src/Tribe/Query.php]: We want to use: `Tribe__Events__Venue::POSTTYPE` and `Tribe__Events__Organizer::POSTTYPE`

File: src/Tribe/Query.php [hunk 3]
```diff
@@ -73,8 +73,18 @@ public static function parse_query( $query ) {
 				$query->set( 'paged', $_REQUEST['tribe_paged'] );
 			}
 
-			// Return early as we don't want to change a post that is not an event.
-			if ( $query->is_main_query() && $query->is_single() && $query->get( 'post_type' ) !== Tribe__Events__Main::POSTTYPE )  {
+			// Return early as we don't want to change a post that is not part of the valid group of event post types.
+			$valid_post_types = [
```
Comment (bordoni) [inline src/Tribe/Query.php]: We cannot use `[]` for array on this version of PHP

**Comment by mitogh** [comment]
Ready for another set of eyes, thanks @bordoni

**Pull request merged**